### PR TITLE
osmocli: migrate to osmocli tests in tokenfactory cli

### PR DESCRIPTION
**Pull request by RusAkh**
- refactor query.go
- add test case for GetCmdDenomAuthorityMetadata
- test case for GetCmdDenomsFromCreator
- add field to QueryCliTestCase to check for wrong responses (does not return error)
- refactor RunQueryTestCase to check ExpectedNotEqual
- missed return
- add wrong test cases

## What is the purpose of the change
Migrate to use osmocli tests in tokenfactory module. Also adds a field to `QueryCliTestCase` to check the return value of a query, to be able to check for wrong response (might be not really important to add this, will wait for review)

## Documentation and Release Note
not documented

### CODE REVIEW

**Comment by ValarDragon** [comment]
Tokenfactory code change looks great to me!

> Also adds a field to QueryCliTestCase to check the return value of a query, to be able to check for wrong response (might be not really important to add this, will wait for review)

This doesn't make sense to me, I think we should revert it. This is because the guarantee that "this command builds exactly this proto message", is a much stronger guarantee than "this command builds a proto message that is not equal to {this}". (And the former already implies the latter!)

File: osmoutils/osmocli/cli_tester.go [hunk 1]
```diff
@@ -21,9 +21,10 @@ type TxCliTestCase[M sdk.Msg] struct {
 }
 
 type QueryCliTestCase[Q proto.Message] struct {
-	Cmd           string
-	ExpectedQuery Q
-	ExpectedErr   bool
+	Cmd              string
+	ExpectedQuery    Q
+	ExpectedErr      bool
+	ExpectedNotEqual bool
```
[the remark this answers is not in the document]
Comment (RusAkh) [reply osmoutils/osmocli/cli_tester.go:27]: yeah that might be completely redundant, I will remove that

**Comment by RusAkh** [comment]
@ValarDragon done

**Review by ValarDragon — APPROVED** [review]
Awesome, ty!

**Pull request merged**